numba/core/codegen.py

```diff
@@ -32,9 +32,11 @@ def dump(header, body):
             raise ValueError(msg)
         else:
             from pygments import highlight
-            from pygments.lexers import GasLexer as lexer
+            from pygments.lexers import GasLexer as gas_lexer
+            from pygments.lexers import LlvmLexer as llvm_lexer
             from pygments.formatters import Terminal256Formatter
             from numba.misc.dump_style import by_colorscheme
+            lexer = llvm_lexer if lang is None else gas_lexer
```

Comment:
Given the mistake that's motivating this change, I think that it would be a good idea to require a string to specify the lexer and then select based on that, this to force thinking about the target language.

Reply:
My goal was to provide minimum changes to current implementation. ok, we can make 3 arg required